ValveResourceFormat/IO/MapExtract.cs

```diff
@@ -648,32 +744,39 @@ BaseEntity NewPropStatic(string modelName) => new CMapEntity()
                     SetPropertiesFromFlags(instance, fragmentFlags);
                     SetTintAlpha(instance, new Vector4(tint, alpha));
 
+                    if (UseHammerInstances)
+                    {
+                        //if (drawGroup.Children.Count == 0)
+                        if (drawSelectionSet.SelectionSetData.selectedObjects.Count == 0)
+                        {
+                            // The particular model instance that will repeat
+                            MapDocument.World.Children.Add(instance);
+                            drawSelectionSet.SelectionSetData.selectedObjects.Add(instance);
+                        }
+
+                        //instance = new CMapInstance() { Target = drawGroup };
+                        GetWorldLayerNode(layerIndex, layerNodes).Children.Add(instance);
+                        continue;
+                    }
+
+                    // Keep adding the same prop
+                    MapDocument.World.Children.Add(instance);
+                    drawSelectionSet.SelectionSetData.selectedObjects.Add(instance);
                     continue;
                 }
 
                 if (drawCenters.Length > 0)
                 {
-                    // fragment recentering
+                    // fragment recentering based on bounding box
                     // apply positive vector in the vmap, and negative vector in the vmdl
                     instance.Origin = drawCenters[i];
                 }
 
-                if (instanceGroup is null)
-                {
-                    Debug.Assert(aggregateHasTransforms == false, "aggregate also has instanced transforms");
-
-                    instanceGroup = new CMapGroup
-                    {
-                        Name = "[MultiDraw] " + Path.GetFileNameWithoutExtension(modelName),
-                    };
-
-                    GetWorldLayerNode(layerIndex, layerNodes).Children.Add(instanceGroup);
-                }
-
                 SetPropertiesFromFlags(instance, fragmentFlags);
                 SetTintAlpha(instance, new Vector4(tint, alpha));
 
-                instanceGroup.Children.Add(instance);
+                MapDocument.World.Children.Add(instance);
```

Comment:
Need to respect world layers.. `GetWorldLayerNode(layerIndex, layerNodes).Children.Add`